website/docs/script/type-id.mdx

```diff
@@ -0,0 +1,237 @@
+---
+id: type-id
+title: "Tutorial: Type ID (Upgradable Script)"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import ScriptHeaders from "./ScriptHeaders.tsx";
+import TutorialHeader from "@components/TutorialHeader";
+import ScriptTools from "./_ScriptTools.mdx";
+
+# Tutorial: Type ID (Upgradable Script)
+
+<TutorialHeader
+  time={ScriptHeaders.typeid.time}
+  topics={ScriptHeaders.typeid.topics}
+  tools={ScriptHeaders.typeid.tools}
+/>
+
+Smart contracts are codes that got executed on-chain that nobody can stops. However, sometimes you might want to implement some kind of upgradability for your smart contract in case there are bugs in the source code.
+
+In CKB, there is a commomly used pattern called [Type ID](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#type-id) to leverage the `hash_type` field in CKB's Script structure to make your Script upgradable. We will explorer this idea in the post.
+
+## Write A Unique Type Script
+
+Let's first ask a question: how can we create a type Script that ensures only one live cell in CKB can have that unique type Script? By unique type Script, we mean the whole type Script structure, including `code_hash`, `hash_type` and `script_args`.
+
+Note that this question might seem irrelevant to Script upgradability right now, but please bear with me, we will see how it will contribute to the final solution in CKB.
+
+The answer is quite simple, here's the basic workflow of the Script:
+
+1. Count how many output cells use current type Script, if there's more than one cell with current type Script, returns a failure return code;
+2. Count how many input cells use current type Script, if there's one input cell with current type Script, returns a success return code;
+3. Use CKB syscall to read the first OutPoint in current transaction;
+4. If the OutPoint data read match the `args` part of current type Script, returns a success return code;
+5. Returns a failure return code otherwise.
```

Comment:
```suggestion
1. Count how many output Cells use current Type Script. If there's more than one Cell with the current Type Script, returns a failure return code.
2. Count how many input Cells use the current Type Script, if there's one input Cell with the current Type Script, returns a success return code.
3. Use the CKB syscall to read the first OutPoint in the current transaction.
4. If the OutPoint data read matches the `args` part of the current Type Script, returns a success return code.
5. Returns a failure return code otherwise.
```